geowave-analytics/src/main/java/mil/nga/giat/geowave/analytics/tools/partitioners/AdapterBasedPartitioner.java

```diff
@@ -0,0 +1,246 @@
+package mil.nga.giat.geowave.analytics.tools.partitioners;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+
+import mil.nga.giat.geowave.analytics.parameters.ClusteringParameters;
+import mil.nga.giat.geowave.analytics.parameters.CommonParameters;
+import mil.nga.giat.geowave.analytics.parameters.ParameterEnum;
+import mil.nga.giat.geowave.analytics.tools.ConfigurationWrapper;
+import mil.nga.giat.geowave.analytics.tools.PropertyManagement;
+import mil.nga.giat.geowave.analytics.tools.RunnerUtils;
+import mil.nga.giat.geowave.analytics.tools.dbops.AccumuloAdapterStoreFactory;
+import mil.nga.giat.geowave.analytics.tools.dbops.AdapterStoreFactory;
+import mil.nga.giat.geowave.analytics.tools.partitioners.AdapterBasedPartitioner.AdapterDataEntry;
+import mil.nga.giat.geowave.index.ByteArrayId;
+import mil.nga.giat.geowave.index.dimension.LongitudeDefinition;
+import mil.nga.giat.geowave.index.dimension.NumericDimensionDefinition;
+import mil.nga.giat.geowave.index.sfc.data.BasicNumericDataset;
+import mil.nga.giat.geowave.index.sfc.data.MultiDimensionalNumericData;
+import mil.nga.giat.geowave.index.sfc.data.NumericData;
+import mil.nga.giat.geowave.index.sfc.data.NumericRange;
+import mil.nga.giat.geowave.store.adapter.AdapterPersistenceEncoding;
+import mil.nga.giat.geowave.store.adapter.AdapterStore;
+import mil.nga.giat.geowave.store.adapter.DataAdapter;
+import mil.nga.giat.geowave.store.index.CommonIndexModel;
+
+/**
+ * This class uses the {@link DataAdapter} to decode the dimension fields to be
+ * indexed. Although seemingly more flexible than the
+ * {@link OrthodromicDistancePartitioner}, handling different types of data
+ * entries, the assumption is that each object decode by the adapter provides
+ * the fields required according to the supplied model.
+ * 
+ * The user provides the distances per dimension. It us up to the user to
+ * convert geographic distance into distance in degrees per longitude and
+ * latitude.
+ * 
+ * 
+ */
+public class AdapterBasedPartitioner extends
+		AbstractPartitioner<AdapterDataEntry> implements
+		Partitioner<AdapterDataEntry>,
+		Serializable
```

Comment:
The issue here is AdapterStore is not serializable.  I think there is a more appropriate way to handle this now without forcing an interface change to AdapterStore.  Will adjust without some treatment now and perhaps revisit later.